app/modules/somfy.c

```diff
@@ -0,0 +1,246 @@
+// ***************************************************************************
+// Somfy module for ESP8266 with NodeMCU
+// 
+// Written by Lukas Voborsky, @voborsky
+// based on https://github.com/Nickduino/Somfy_Remote
+// Somfy protocol description: https://pushstack.wordpress.com/somfy-rts-protocol/
+// and discussion: https://forum.arduino.cc/index.php?topic=208346.0
+// 
+// MIT license, http://opensource.org/licenses/MIT
+// ***************************************************************************
+
+//#define NODE_DEBUG
+
+#include "os_type.h"
+#include "osapi.h"
+
+#include "module.h"
+#include "lauxlib.h"
+#include "lmem.h"
+#include "platform.h"
+#include "hw_timer.h"
+#include "user_interface.h"
+
+#define SYMBOL 640 // symbol width in microseconds
+#define SOMFY_UP    0x2
+#define SOMFY_STOP  0x1
+#define SOMFY_DOWN  0x4
+#define SOMFY_PROG  0x8
+
+#define DIRECT_WRITE_LOW(pin)    (GPIO_OUTPUT_SET(GPIO_ID_PIN(pin_num[pin]), 0))
+#define DIRECT_WRITE_HIGH(pin)   (GPIO_OUTPUT_SET(GPIO_ID_PIN(pin_num[pin]), 1))
+
+static const os_param_t TIMER_OWNER = 0x736f6d66; // "somf"
+static task_handle_t done_taskid;
+
+static uint8_t    pin;
+static uint8_t    frame[7];
+static uint8_t    sync;
+static uint8_t    repeat;
+
+//static uint32_t   delay[10] = {9415, 89565, 4*SYMBOL, 4*SYMBOL, 4*SYMBOL, 4550, SYMBOL, SYMBOL, SYMBOL, 30415}; // in us
+static const __attribute__((section(".data.delay"))) uint32_t delay[10] = {US_TO_RTC_TIMER_TICKS(9415), US_TO_RTC_TIMER_TICKS(89565), US_TO_RTC_TIMER_TICKS(4*SYMBOL), US_TO_RTC_TIMER_TICKS(4*SYMBOL), US_TO_RTC_TIMER_TICKS(4*SYMBOL), US_TO_RTC_TIMER_TICKS(4550), US_TO_RTC_TIMER_TICKS(SYMBOL), US_TO_RTC_TIMER_TICKS(SYMBOL), US_TO_RTC_TIMER_TICKS(SYMBOL), US_TO_RTC_TIMER_TICKS(30415)}; // in ticks (no need to recalulate)
```

Comment:
I'd even be inclined to add this to `sections.h` and use that instead:
`#define RAM_CONST_SECTION_ATTR __attribute((section((".data")))`

Since it's a static in a module there's nothing to be gained from giving it a separately named section I believe, and it would add a bit of clarity to use a #define instead.